### PR TITLE
[feat]: [CCM-7623]: NG License API in batch processing 

### PR DESCRIPTION
**Pull request by hitesharinga**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/33079)

### CODE REVIEW

File: 930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java [hunk 1]
```diff
@@ -306,6 +306,20 @@ public ResponseDTO<Boolean> softDelete(@PathParam("accountId") String accountId)
...
 
+  @GET
+  @Path("{{accountId}/accenabled")
+  @ApiOperation(
+      value = "Get Module Licenses for Specific Module Type which Expiry Time is Greater or Equal to the Provided Time",
+      nickname = "getModuleLicenseForSpecificModuleType", hidden = true)
+  public ResponseDTO<List<ModuleLicenseDTO>>
+  getAccModuleLicensesByModuleType(@Parameter(required = true, description = "A Harness Platform module.") @NotNull
+                                   @PathParam("accountId") String accountId,
+      @Parameter(
```
Comment (hitesharinga) [inline 930-ng-license-manager/src/main/java/io/harness/licensing/api/resource/LicenseResource.java:317]: remove this

File: 930-ng-core-clients/src/main/java/io/harness/utils/RestCallToNGManagerClientUtils.java [hunk 2]
```diff
@@ -31,6 +31,8 @@ public static <T> T execute(Call<ResponseDTO<T>> request) {
         log.info("response raw {}, body {}", response.raw(), response.body().getData());
         return response.body().getData();
       } else {
+        log.error("request {}", request.request());
```
Comment (hitesharinga) [inline 930-ng-core-clients/src/main/java/io/harness/utils/RestCallToNGManagerClientUtils.java:34]: remove this

**Comment by hitesharinga** [comment]
trigger messagecheck

File: 280-batch-processing/src/main/java/io/harness/batch/processing/shard/AccountShardService.java [hunk 3]
```diff
@@ -53,19 +73,61 @@ public List<Account> getCeEnabledAccounts() {
...
+  private List<Account> getCgAccounts() {
+    return cloudToHarnessMappingService.getCeEnabledAccounts();
+  }
+
+  private List<ModuleLicenseDTO> getNgAccounts() {
+    long expiryTime = Instant.now().minus(15, ChronoUnit.DAYS).toEpochMilli();
+    try {
+      Call<ResponseDTO<List<ModuleLicenseDTO>>> moduleLicensesByModuleType =
+          ngLicenseHttpClient.getModuleLicensesByModuleType(ModuleType.CE, expiryTime);
+      return RestCallToNGManagerClientUtils.execute(moduleLicensesByModuleType);
```
[the remark this answers is not in the document]
Comment (nikunj-badjatya) [reply 280-batch-processing/src/main/java/io/harness/batch/processing/shard/AccountShardService.java:124]: Thoughts on this @hitesharinga , @svmav ?

**Review by nikunj-badjatya — CHANGES_REQUESTED** [review]

File: 280-batch-processing/src/main/java/io/harness/batch/processing/shard/AccountShardService.java (hunk 3 above)
[the remark this answers is not in the document]
Comment (hitesharinga) [reply 280-batch-processing/src/main/java/io/harness/batch/processing/shard/AccountShardService.java:124]: Sounds good. I will add that. 
Right now its both NG manager and CG, I can add that when we will remove CG.

[the remark this answers is not in the document]
Comment (nikunj-badjatya) [reply 280-batch-processing/src/main/java/io/harness/batch/processing/shard/AccountShardService.java:124]: cool

**Review by nikunj-badjatya — COMMENTED** [review]
> Reviewed 17 of 19 files at r1, 2 of 2 files at r2, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33079)* status: Some of the checks are too old. 1 change requests. 1. unresolved discussions. 0 of 1 approvals obtained.

**Review by nikunj-badjatya — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/33079)* status: Some of the checks are too old.

**Pull request merged**